src/price/price.service.ts

```
@@ -1,29 +1,61 @@
-import { Injectable } from '@nestjs/common'
-import { PrismaService } from 'src/prisma/prisma.service'
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { PrismaService } from '../prisma/prisma.service'
 import { priceDto } from './dto'
 
 @Injectable()
 export class PriceService {
   constructor(private prisma: PrismaService) {}
-  async findOrCreate(data: priceDto) {
-    return this.prisma.price.upsert({
+
+  async findOrCreate(priceData: priceDto) {
+    const { bookId, platformId, date, productUrl } = priceData
+
+    let price = await this.prisma.price.findFirst({
       where: {
-        bookId_platformId: {
-          bookId: data.bookId,
-          platformId: data.platformId,
+        bookId,
+        platformId,
+        date,
+        productUrl,
+      },
+    })
+
+    if (!price) {
+      price = await this.prisma.price.create({
+        data: {
+          bookId,
+          platformId,
+          price: priceData.price,
+          date,
+          productUrl,
         },
+      })
+    }
+    return price
+  }
+
+  async updatePrice(id: string, priceData: priceDto) {
+    const existingPrice = await this.prisma.price.findUnique({
+      where: {
+        id: parseInt(id),
       },
-      update: {
-        price: data.price,
-        date: data.date,
+    })
+    if (!existingPrice) {
+      throw new NotFoundException(`Price with ID ${id} not found`)
+    }
+    return this.prisma.price.update({
+      where: {
+        id: existingPrice.id,
       },
-      create: {
-        bookId: data.bookId,
-        platformId: data.platformId,
-        price: data.price,
-        date: data.date,
-        productUrl: data.productUrl,
+      data: {
+        price: priceData.price,
       },
     })
   }
+
+  async deletePrice(id: number) {
+    const deletedPrice = await this.prisma.price.update({
```

If the id has not been found, it should raise an exception